BaragonAgentService/src/main/java/com/hubspot/baragon/agent/lbs/LbConfigGenerator.java

```diff
@@ -11,48 +17,59 @@
 import com.hubspot.baragon.agent.BaragonAgentServiceModule;
 import com.hubspot.baragon.agent.config.LoadBalancerConfiguration;
 import com.hubspot.baragon.agent.models.LbConfigTemplate;
+import com.hubspot.baragon.exceptions.MissingTemplateException;
 import com.hubspot.baragon.models.BaragonConfigFile;
+import com.hubspot.baragon.models.BaragonService;
 import com.hubspot.baragon.models.ServiceContext;
 
 @Singleton
 public class LbConfigGenerator {
   private final LoadBalancerConfiguration loadBalancerConfiguration;
-  private final Collection<LbConfigTemplate> templates;
-  
+  private final Map<String, List<LbConfigTemplate>> templates;
+
   @Inject
   public LbConfigGenerator(LoadBalancerConfiguration loadBalancerConfiguration,
-                           @Named(BaragonAgentServiceModule.AGENT_TEMPLATES) Collection<LbConfigTemplate> templates) {
+                           @Named(BaragonAgentServiceModule.AGENT_TEMPLATES) Map<String, List<LbConfigTemplate>> templates) {
     this.loadBalancerConfiguration = loadBalancerConfiguration;
     this.templates = templates;
   }
 
-  public Collection<BaragonConfigFile> generateConfigsForProject(ServiceContext snapshot) {
-    final Collection<BaragonConfigFile> files = Lists.newArrayListWithCapacity(templates.size());
+  public Collection<BaragonConfigFile> generateConfigsForProject(ServiceContext snapshot) throws MissingTemplateException {
+    final Collection<BaragonConfigFile> files = Lists.newArrayList();
+    String templateName = snapshot.getService().getTemplate() != null ? snapshot.getService().getTemplate() : "default";
```

Comment:
consider making the `template` field optional -- then we can just do `getTemplate().or("default")`